internal/engine/sprite.go

```diff
@@ -1,25 +1,21 @@
 package engine
 
 import (
-	. "github.com/realdream-ai/gdspx/pkg/engine"
+	"github.com/goplus/spx/internal/enginewrap"
+	gdx "github.com/realdream-ai/gdspx/pkg/engine"
 	. "github.com/realdream-ai/mathf"
 )
 
-type ProxySprite struct {
-	Sprite
+// !!!Warning all method belong to this class can only be called in main thread
```

Comment:
Nit: To ensure developers always remember this when manipulating `Sprite`, we may use a specific prefix or suffix for related methods, such as `UpdateTexture_MustFromMainThread`. While it may look ugly, it could be beneficial. 🤣